packages/graphql/src/contracts.js

```diff
@@ -62,16 +63,23 @@ export function newBlock(blockHeaders) {
 }
 
 function pollForBlocks() {
+  let inProgress = false
   try {
     blockInterval = setInterval(() => {
+      if (inProgress) {
+        return
+      }
+      inProgress = true
       web3.eth.getBlockNumber().then(block => {
```

Comment:
Should we have a .catch statement hat logs an error and sets inProgress to false ?
To handle the case where a call to web3.eth.getBlockNumber fails.